src/transforms/levels.jl

```diff
@@ -0,0 +1,72 @@
+"""
+    Levels(:a => ["yes,"no"])
+
+Return a copy of the table with specified levels and orders for categorical columns
+allowing only changing the order of the column.
+
+# Examples
+
+```julia
+Levels(:a => ["yes, "no"], :c => [1, 2, 4], :d => ["a", "b", "c"])
+Levels("a" => ["yes", "no"], "c" => [1, 2, 4], ordered = ["a", "c"])
+Levels(:a => ["yes", "no"], :c => [1, 2, 4], :d => [1, 23, 5, 7], ordered = [:a, :b, :c])
+```
+"""
+struct Levels{K} <: Stateless
+  levelspec::K
+  ordered::Vector{Symbol}
+end
+
+Levels(pairs::Pair{Symbol}...; ordered=Symbol[]) =
+  Levels(NamedTuple(pairs), ordered)
+
+Levels(pairs::Pair{K}...; ordered=K[]) where {K<:AbstractString} =
+  Levels(NamedTuple(Symbol(k) => v for (k,v) in pairs), Symbol.(ordered))
```

Comment:
The `Levels` constructor will break in this case:
```julia
julia> Levels(:a => ["yes", "no"], :c => [1, 2, 4], :d => [1, 23, 5, 7], ordered = ["a", "b"])
ERROR: MethodError: no method matching Levels(::NamedTuple{(:a, :c, :d), Tuple{Vector{String}, Vector{Int64}, Vector{Int64}}}, ::Vector{String})
Closest candidates are:
  Levels(::K, ::Vector{Symbol}) where K
```
This is a simple solution to this problem:
```suggestion
Levels(levelspec::NamedTuple, ordered::Vector{K}) where {K<:AbstractString} =
  Levels(levelspec, Symbol.(ordered))

Levels(pairs::Pair{Symbol}...; ordered=Symbol[]) =
  Levels(NamedTuple(pairs), ordered)

Levels(pairs::Pair{K}...; ordered=K[]) where {K<:AbstractString} =
  Levels(NamedTuple(Symbol(k) => v for (k,v) in pairs), ordered)
```

Reply:
Yes, but I guess users would be consistent if they use Symbol in the first place, they would do so for the second one. Do you think it would be better to add the functionality to support it because I don't think to mix `Symbol` and `String` in one expression is good in any use case. Could you talk more about it? (The suggestion is very nice to me since I am so struggled with constructors)

Reply:
This change is just to make the `Levels` constructor more generic, as in some cases this may be necessary.

Reply:
I agree with @Frank-III that the use case with mixed types is not very welcome. @eliascarv can we simply leave the PR as is without this fuzz with mixed Symbol/String arguments? We need to think in terms of what we want to support and maintain in the long-term. I don't think we want to maintain mixed types in this case, right?

Reply:
Note that this works:
```julia
Levels("a" => ["yes", "no"], "c" => [1, 2, 4], "d" => [1, 23, 5, 7], ordered = [:a, :b])
```
So it's weird that the opposite doesn't work.

Reply:
Yes, the fact that it works as a side effect of the current implementation doesn't mean that we want to support it officially. If you intentionally write constructors to support mixed types, that means that in future releases we will need to submit more fixes to other transforms that operate with Symbol/String in the same function call.